apps/loader/loader.js

```diff
@@ -109,6 +110,49 @@ function handleUpload(file, filename) {
   );
 }
 
+function handleDownload(id) {
+  var fileName='';
+  store.getSlide(id)
+      .then((response) => {
+        if (response[0]) {
+          return response[0]['location'];
+        } else {
+          throw new Error('Slide not found');
+        }
+      }).then((location) => {
+        fileName= location.substring(8, location.length);
```

Comment:
wouldn't it make more sense to split on "/" instead?

Reply:
Yes, of course. 
I'm on it.

Reply:
Looks right now?